src/HeaderPlugin.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Manages http headers.
+ */
+class HeaderPlugin implements Plugin
+{
+    /**
+     * @var array headers that have fixed value
+     */
+    private $headers = [];
+    private $removedHeader = [];
+
+    /**
+     * Sets a header with the given value to every requests.
+     *
+     * @param string $name     name of the header
+     * @param string $value    value of the header
+     * @param bool   $override By default if the header is already set on the request the value wont be replaced
+     *                         pass $override to true to make the header value the same for every request
+     */
+    public function withHeader($name, $value, $override = false)
```

Comment:
I am not sure about this method:

1. with prefix suggests that it is immutable, while it is not.
2. I see the point in a builder function, but plugins should be configured once and never touched again. Although requireing an array of header removals is probably harder (or not from configuration?),

/cc @joelwurtz